### DISCUSSION TITLE
Error with Sequelize import on the client side

### DISCUSSION THREAD

**Discussion by jamuna-r7o** [vercel/next.js#15323] [Help]
This question is closely related to [Issues with multi-database libs: sequelize #345](https://github.com/vercel/ncc/issues/345#issuecomment-487404520)

## Problem
I am trying to import my database on a page to use it for `getServerSideProps` to get data for the page. However, the Sequelize initialisation is failing on any page that is not an API route. I need to use it in `getServerSideProps` otherwise I won't be able to query my database and get the data I need for the page.

## **pages/messages/index.js**
The root of the error. The imported middleware contains all the middlewares like passport.js, and database.js.
`
import middleware from 'middlewares/middleware';
`

## **middlewares/database.js**
This contains the sequelize initialisation. Importantly, this is where the code fails on the client side, but it works in an API route.
```js
const sequelize = new Sequelize(process.env.DB_NAME, process.env.DB_USER, process.env.DB_PASS, {
  host: process.env.DB_HOST,
  dialect: 'postgres',
});
```

This is the error message:
**Unhandled Runtime Error
Error: Please install pg package manually**

Others have advised that adding `dialectModule` have fixed it for them, but with the value `'pg' `it gave me this error: **`Cannot find module 'pg-native'`**

```js
const sequelize = new Sequelize(process.env.DB_NAME, process.env.DB_USER, process.env.DB_PASS, {
  host: process.env.DB_HOST,
  dialect: 'postgres',
  dialectModule: require('pg'),
});
```

So I tried to install pg-native but I first got the error **`Can't find Python executable "python"`** so I had to add python to an environmental variable, but then I got another error **`'pg_config' is not recognized as an internal or external command`**, so I had to install `production windows-build-tools` and `node-gyp` to fix this:

```
npm npm install --global --production windows-build-tools
npm install node-gyp
```

It turns out `pg-native` just gives me a whole new set of bugs, so I ditched it.

I'm getting this error message now `Module not found: Can't resolve 'dns' ` when I try to import sequelize from a non API route. I tried to fake this package as it's to do with promises, but then it could not find `fs`.

**Comment by jamuna-r7o** [comment]
It is closely related to this issue
https://github.com/vercel/now-builders/issues/331

**Comment by jamesmosier** [comment]
Have you tried installing [`pg`](https://www.npmjs.com/package/pg)?

**Comment by jamuna-r7o** [comment]
Yes, it has caused an endless spiral of errors. Adding `dialectModule` was supposed to fix it, but with `'pg' `it's giving me this error: **`Cannot find module 'pg-native'`**

```
const sequelize = new Sequelize(process.env.DB_NAME, process.env.DB_USER, process.env.DB_PASS, {
  host: process.env.DB_HOST,
  dialect: 'postgres',
  dialectModule: require('pg'),
});
```

So I'm trying to install `pg-native` but I first got the error **`Can't find Python executable "python"`** so I had to add python to an environmental variable, but now I'm getting another error **`'pg_config' is not recognized as an internal or external command`**.

**Comment by jamuna-r7o** [comment]
Getting these errors now

**`ModuleNotFoundError: Module not found: Error: Can't resolve 'dns'`**
